sql/index.go

```diff
@@ -49,6 +49,8 @@ type Index interface {
 	// ColumnExpressionTypes returns each expression and its associated Type. Each expression string should exactly
 	// match the string returned from Index.Expressions().
 	ColumnExpressionTypes(ctx *Context) []ColumnExpressionType
+	// ColumnNames returns the names of the Columns that this Index is defined over
+	ColumnNames() []string
```

Comment:
This method is effectively a duplicate of `Expressions() []string`. Instead of adding `ColumnNames()`, you should use the `Expressions()` in combination with [`GetColumnFromIndExpr`](https://github.com/dolthub/go-mysql-server/blob/e9245f8a2755855655c443463bcab173603cdb1d/sql/plan/show_indexes.go#L162)